example_remoteSign_test.go

```diff
@@ -0,0 +1,68 @@
+package notation_test
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+
+	_ "github.com/notaryproject/notation-core-go/signature/cose"
+	_ "github.com/notaryproject/notation-core-go/signature/jws"
+	"github.com/notaryproject/notation-core-go/testhelper"
+	"github.com/notaryproject/notation-go"
+	"github.com/notaryproject/notation-go/internal/mock"
+	"github.com/notaryproject/notation-go/signer"
+)
+
+// ExampleRemoteSign demonstrates how to use notation.Sign to sign an artifact
+// in the remote registry and push the signature to the remote.
+func Example_remoteSign() {
+	exampleArtifactReference := "registry.acme-rockets.io/software/net-monitor@sha256:60043cf45eaebc4c0867fea485a039b598f52fd09fd5b07b0b2d2f88fad9d74e"
+
+	// Both COSE ("application/cose") and JWS ("application/jose+json")
+	// signature mediaTypes are supported.
+	exampleSignatureMediaType := "application/cose"
+
+	// exampleCertTuple contains a RSA privateKey and a self-signed X509
+	// certificate generated for demo purpose ONLY.
+	// Users should bring their own full certificate chain following the
+	// Notary certificate requirements:
+	// https://github.com/notaryproject/notaryproject/blob/v1.0.0-rc.1/specs/signature-specification.md#certificate-requirements
+	exampleCertTuple = testhelper.GetRSASelfSignedSigningCertTuple("Notation Example self-signed")
+	exampleCerts = []*x509.Certificate{exampleCertTuple.Cert}
+
+	// exampleSigner is a notation.Signer given key and X509 certificate chain.
+	exampleSigner, err := signer.New(exampleCertTuple.PrivateKey, exampleCerts)
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	// exampleRepo is a dummy registry.Repository for demo purpose ONLY.
+	// Users are recommended to use registry.NewRepository() for implementation
+	// of registry.Repository. (https://github.com/notaryproject/notation-go/blob/v1.0.0-rc.1/registry/repository.go#L25)
+	exampleRepo := mock.NewRepository()
```

Comment:
I think you can reference `oras` here instead of a mock.

Reply:
It is OK that the example is not runnable. See examples from built-in libraries: https://pkg.go.dev/net#example-Dialer.

/cc @Wwwsylvia for `oras-go`.